### ISSUE TITLE
Configure how clangd is called to remove --driver-mode

### ISSUE THREAD

**Issue by kheaactua** [neovim/nvim-lspconfig#921]
(not sure if this is the right place to ask this)

I see in my `lsp.log` that my compiler is being called with `--driver-mode=g++` and `--resource-dir`.  These flags make sense when I read about them, but I'm using a QNX cross compiler (`ntoaarch64-g++`) that doesn't support these flags.

My set up is simply:
```lua
lspconfig.clangd.setup{
  cmd = { vim.g.clang_path .. "/bin/clangd", "--background-index" }
}
```

The compile commands are extracted from my `compile_commands.json` file generated by CMake.

Is there any way to modify my setup to not add these two flags?  Basically when I input the failed command from the log, the command shows that it fails immediately.

**Comment by mjlbach** [comment]
Usage questions belong on our discourse. If I had to guess, clangd is reading these flags directly from the compile_commands.json, so you'd need to modify that file. I'd read the clangd documentation to see if that can be overridden by an init option or settings that we can pass when the server starts

**Comment by kheaactua** [comment]
Hi, I wasn't aware of the discourse, I'll check that out.

But no, the commands aren't in the `compile_commands.json` file.

The target in `compile_commands.json` is:
```json
{
  "directory": "/home/matt/workspace/capi/tmp/build-hlos/vsomeip",
  "command": "/opt/hlos/qnxsdp-qualcomm/host/linux/x86_64/usr/bin/ntoaarch64-g++ -DVSOMEIP_VERSION=\\\"3.1.20.3\\\" -Dvsomeip3_EXPORTS -I/home/matt/workspace/capi/tmp/vsomeip/interface -isystem /home/matt/workspace/capi/tmp/vsomeip/implementation/helper/1.66 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -D_FORTIFY_SOURCE=2 -Os -fstack-protector-strong -fstack-protector-strong -D_XOPEN_SOURCE=600 -DNI_MAXHOST=1025 -DNI_MAXSERV=32 -DVSOMEIP_ENABLE_SIGNAL_HANDLING -DWITHOUT_SYSTEMD -g -std=c++11 -DNDEBUG -fPIC -o CMakeFiles/vsomeip3.dir/implementation/endpoints/src/endpoint_impl.cpp.obj -c /home/matt/workspace/capi/tmp/vsomeip/implementation/endpoints/src/endpoint_impl.cpp",
  "file": "/home/matt/workspace/capi/tmp/vsomeip/implementation/endpoints/src/endpoint_impl.cpp"
},
```

The log output shows:
```
[ ERROR ] 2021-05-20T13:46:33-0400 ] /usr/local/share/nvim/runtime/lua/vim/lsp/rpc.lua:458 ]	"rpc"	"/usr/bin/clangd"	"stderr"	'I[13:46:33.163] ASTWorker building file /home/matt/workspace/capi/tmp/vsomeip/implementation/endpoints/src/endpoint_impl.cpp version 0 with command \n[/home/matt/workspace/capi/tmp/build-hlos/vsomeip]\n/opt/hlos/qnxsdp-qualcomm/host/linux/x86_64/usr/bin/ntoaarch64-g++ --driver-mode=g++ -DVSOMEIP_VERSION="3.1.20.3" -Dvsomeip3_EXPORTS -I/home/matt/workspace/capi/tmp/vsomeip/interface -isystem /home/matt/workspace/capi/tmp/vsomeip/implementation/helper/1.66 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -D_FORTIFY_SOURCE=2 -Os -fstack-protector-strong -fstack-protector-strong -D_XOPEN_SOURCE=600 -DNI_MAXHOST=1025 -DNI_MAXSERV=32 -DVSOMEIP_ENABLE_SIGNAL_HANDLING -DWITHOUT_SYSTEMD -DQNX -g -std=c++11 -DNDEBUG -fPIC -o CMakeFiles/vsomeip3.dir/implementation/endpoints/src/endpoint_impl.cpp.obj -c /home/matt/workspace/capi/tmp/vsomeip/implementation/endpoints/src/endpoint_impl.cpp -fsyntax-only -resource-dir=/usr/lib/llvm-11/lib/clang/11.1.0\n'
```

It's possible (likely) my toolchain file isn't setup properly, but the problem at hand is that the command fails immidiately due to the `--driver-mode=g++` and `-resource-dir` flags shown in the log that are no where to be found in any `compile_commands.json` file.

**Comment by mjlbach** [comment]
I'm assuming it's this https://reviews.llvm.org/D62804

We're not setting driver-mode in lspconfig, if you can find a setting/initOption that can be passed to clangd on initialization, you can set it in your personal settings override. I'd recommend:

* Browsing the options at https://clangd.llvm.org/config.html
* Asking on the discourse at https://llvm.discourse.group/c/llvm-project/clangd/34
* Asking on the clang discord at https://discord.com/invite/xS7Z362
* And if you believe it's a bug, filing at https://github.com/clangd/clangd/issues

Unfortunately, there isn't much we can do, as even if this *is* a setting that we could change in the defaults. it's probably specific to your usecase/project.

**Comment by kheaactua** [comment]
Thanks for the help @mjlbach!

**Comment by kheaactua** [comment]
Actually, I think I did solve the issue.  Posting here in case someone else has a similar issue in the future.

No idea why I didn't find it last time, but reading deeper into clangd I found that I could create a config file, so in the base of my project I added `.clangd`

```
CompileFlags:
  Remove: [--driver-mode=*, -resource-dir=*]
```

Simple, and it appears to be working.  Now I just don't understand how I found it after 2 minutes today, but didn't after an hour of searching on Thursday...

Anyways, thanks again for the help!
